### PR TITLE
Metric feature tests plus some refactoring

### PR DESCRIPTION
**Pull request by kapilra**

### CODE REVIEW

File: timeseries-aggregator/src/main/scala/com/expedia/www/haystack/metricpoints/entities/MetricPoint.scala [hunk 1]
```diff
@@ -37,9 +36,18 @@ case class MetricPoint(metric: String,
 object MetricType extends Enumeration {
   type MetricType = Value
   val Metric, Histogram, Aggregate = Value
-
 }
 
+object HistogramType extends Enumeration {
```
Comment (ayansen) [inline timeseries-aggregator/src/main/scala/com/expedia/www/haystack/metricpoints/entities/MetricPoint.scala]: lets rename this to HistogramStats

File: timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  val keys = Map(TagKeys.OPERATION_NAME_KEY -> OPERATION_NAME,
+    TagKeys.SERVICE_NAME_KEY -> SERVICE_NAME)
+
+  feature("Creating a Metric (Count / Histogram") {
+    scenario("should get Histogram MetricType and HistogramTypes for duration MetricPoints") {
+
+      Given("some duration MetricPoints")
+      val duration1: Long = 10
```
Comment (ayansen) [inline timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala]: use list of durations and use that to construct list of metricpoints -> durations = List(10,140)
val metricPoints = durations.map(duration => MetricPoint(DURATION_METRIC_NAME, MetricType.Metric, keys, duration, System.currentTimeMillis))

File: timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+  object TagKeys {
+    val OPERATION_NAME_KEY = "operationName"
+    val SERVICE_NAME_KEY = "serviceName"
+  }
+
+  val keys = Map(TagKeys.OPERATION_NAME_KEY -> OPERATION_NAME,
+    TagKeys.SERVICE_NAME_KEY -> SERVICE_NAME)
+
+  feature("Creating a Metric (Count / Histogram") {
```
Comment (ayansen) [inline timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala]: is it a common feature test for metric factory and metric rule engine? I think  metric factory and metric rule engine should be treated as different features. the metric rule engine feature tests should just talk about fetching the right metric type, and the metric factory should talk about fetching the right metric given a metric type, so we can break this test class into two classes

Comment (kapilra) [inline timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala]: We already have tests for MetricRuleEngine. This one is just for MetricFactory.

File: timeseries-aggregator/src/main/scala/com/expedia/www/haystack/metricpoints/entities/MetricPoint.scala (hunk 1 above)
Comment (kapilra) [inline timeseries-aggregator/src/main/scala/com/expedia/www/haystack/metricpoints/entities/MetricPoint.scala]: Done

File: timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala (hunk 2 above)
Comment (kapilra) [inline timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala]: Done

**Pull request merged**